app/slave/cluster_slave.py

```diff
@@ -101,21 +104,23 @@ def setup_build(self, build_id, project_type_params):
         SafeThread(
             target=self._async_setup_build,
             name='Bld{}-Setup'.format(build_id),
-            args=(executors, project_type_params)
+            args=(executors, project_type_params, num_executors_already_allocated)
         ).start()
 
-    def _async_setup_build(self, executors, project_type_params):
+    def _async_setup_build(self, executors, project_type_params, num_executors_already_allocated):
         """
         Called from setup_build(). Do asynchronous setup for the build so that we can make the call to setup_build()
         non-blocking.
 
         :type executors: list[SubjobExecutor]
         :type project_type_params: dict
+        :type num_executors_already_allocated: int
         """
-        # todo(joey): It's strange that the project_type is setting up the executors, which in turn set up projects.
-        # todo(joey): I think this can be untangled a bit -- we should call executor.configure_project_type() here.
+        self._base_executor_index = num_executors_already_allocated
         try:
-            self._project_type.fetch_project(executors, project_type_params)
+            for executor in executors:
+                executor.configure_project_type(project_type_params)
+            self._project_type.fetch_project()
```

Comment:
I believe the executor configuration should occur after the project has been fetched.